### PR TITLE
AAP-36992: ui-test: isolate each tests

### PR DESCRIPTION
**Pull request by goneri**
- Isolate each test with a different `extest` process.
- Stop the UI tests as soon as one test has failed.
- Split up `lightspeedUiTest.ts` in 4 different files.
- Collect the Selenium screenshots.
- Reduce the number of Selenium interactions by putting the settings in `settings.json` directly.
- A couple of extra simplifications.
- Don't test the SignOut. This is under VSCode control and there is no point testing it.

### CODE REVIEW

**Comment by manstis** [comment]
@goneri let me know if you need some eyes on this when ready for review 👍

**Comment by goneri** [comment]
@manstis thank you. It should be ready for review soon.

**Comment by manstis** [comment]
Oh no @goneri I've broken your PR.

The merge conflict _should_ be straight forward; but if you'd prefer me to rollback my [PR](https://github.com/ansible/vscode-ansible/pull/1689) let me know.

**Comment by goneri** [comment]
> Oh no @goneri I've broken your PR.
> 
> The merge conflict _should_ be straight forward; but if you'd prefer me to rollback my [PR](https://github.com/ansible/vscode-ansible/pull/1689) let me know.

Oh. Can we revert it the time we land this PR?

**Comment by goneri** [comment]
@ssbarnea @alisonlhart @manstis  I know about the codecov regression. I'm not totally sure where it is coming from yet, but at this stage the PR is getting really big and I would prefer to land it and iterate.

File: test/ui-test/lightspeedAuthUiTest.ts [hunk 1]
```diff
@@ -18,146 +18,145 @@ import {
...
-      // Enable Lightspeed and open Ansible Light view on sidebar
-      workbench = new Workbench();
-      const settingsEditor = await workbench.openSettings();
-      await updateSettings(settingsEditor, "ansible.lightspeed.enabled", true);
-      await updateSettings(
-        settingsEditor,
-        "ansible.lightspeed.URL",
-        process.env.TEST_LIGHTSPEED_URL,
-      );
+before(function () {
```
Comment (manstis) [inline test/ui-test/lightspeedAuthUiTest.ts:21]: Are these disabled?

**Review by manstis — COMMENTED** [review]
Great job. Perhaps running tests will not be so fearsome in the future!

I'm however unable to run the tests locally to verify things work:
```
Launching tests...
  Verify the execution of playbook using ansible-playbook command
    ✔ Execute ansible-playbook command with arg (19187ms)
    ✔ Execute ansible-playbook command without arg (15926ms)

  Verify the execution of playbook using ansible-navigator command
    1) Execute playbook with ansible-navigator EE mode
    2) Execute playbook with ansible-navigator without EE mode
```
Tests using `ansible-navigator` have *always* failed for me but I've been able to ignore them as the rest of the test suite ran. However this PR introduces the concept of "fail fast" and now I cannot run the whole suite.

How/what do I need to configure to get the `ansible-navigator` tests to pass? I have `ansible-navigator` installed and typing the same at the CLI launches something.

I'd like to run tests before approving.

File: test/ui-test/lightspeedAuthUiTest.ts (hunk 1 above)
Comment (goneri) [reply test/ui-test/lightspeedAuthUiTest.ts:21]: Hum, they should be enabled only on MacOS. Let's me change that.

**Comment by goneri** [comment]
> Great job. Perhaps running tests will not be so fearsome in the future!
> 
> I'm however unable to run the tests locally to verify things work:
> 
> ```
> Launching tests...
>   Verify the execution of playbook using ansible-playbook command
>     ✔ Execute ansible-playbook command with arg (19187ms)
>     ✔ Execute ansible-playbook command without arg (15926ms)
> 
>   Verify the execution of playbook using ansible-navigator command
>     1) Execute playbook with ansible-navigator EE mode
>     2) Execute playbook with ansible-navigator without EE mode
> ```
> 
> Tests using `ansible-navigator` have _always_ failed for me but I've been able to ignore them as the rest of the test suite ran. However this PR introduces the concept of "fail fast" and now I cannot run the whole suite.
> 
> How/what do I need to configure to get the `ansible-navigator` tests to pass? I have `ansible-navigator` installed and typing the same at the CLI launches something.
> 
> I'd like to run tests before approving.

Those are also broken for my but I'm also not too surprised because I use Fedora Silverblue (`podman` has to be called with `--remote` to go through the socket).

**Review by manstis — APPROVED** [review]
LGTM :+1: 

I'll worry about the `ansible-navigator` tests later (even if I have to disable them locally).

I normally run `task test-ui` locally before pushing.

**Pull request merged**